src/modules/requirements/GymBadgeRequirement.ts

```diff
@@ -14,6 +14,8 @@ export default class GymBadgeRequirement extends Requirement {
     }
 
     public hint(): string {
-        return `Requires the ${GameConstants.camelCaseToString(BadgeEnums[this.badge])} badge.`;
+        return BadgeEnums[this.badge].startsWith('Elite') ?
```

Comment:
```suggestion
        return BadgeEnums[this.badge].includes('Elite') || BadgeEnums[this.badge].includes('Champion') ?
```
Because Champion Hau's badge is named Champion_Stamp.

Reply:
This PR is for more helpful badge hints. The Champion Stamp is visible in the Badge Case, so it doesn't need a different hint. We denote "badgeless" enums with the `Elite_` prefix, which is what this line targets.